pymc3/sampling.py

```diff
@@ -419,6 +419,29 @@ def sample(
 
     """
     model = modelcontext(model)
+    if start is None:
+        start = model.test_point
+    else:
+        if isinstance(start, dict):
+            update_start_vals(start, model.test_point, model)
+        else:
+            for chain_start_vals in start:
+                update_start_vals(chain_start_vals, model.test_point, model)
+
+    start_points = [start] if isinstance(start, dict) else start
+    for elem in start_points:
+        if not set(elem.keys()).issubset(model.named_vars.keys()):
+            extra_keys = ", ".join(set(elem.keys()) - set(model.named_vars.keys()))
+            valid_keys = ", ".join(model.named_vars.keys())
+            raise KeyError("Some start parameters do not appear in the model!\n"
+                           "Valid keys are: {}, but {} was supplied".format(valid_keys, extra_keys))
+
+        initial_eval = model.check_test_point(test_point=elem)
+
+        if not np.all(np.isfinite(initial_eval)):
```

Comment:
Feel free to ignore if you don't agree it is more readable.
```suggestion
    if np.any(~np.isfinite(initial_eval)):
```